Mage.Sets/src/mage/cards/c/CursedCourtier.java

```diff
@@ -0,0 +1,43 @@
+package mage.cards.c;
+
+import mage.MageInt;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.effects.common.CreateRoleAttachedSourceEffect;
+import mage.abilities.keyword.LifelinkAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.RoleType;
+import mage.constants.SubType;
+
+import java.util.UUID;
+
+/**
+ * @author TheElk801
+ */
+public final class CursedCourtier extends CardImpl {
+
+    public CursedCourtier(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{W}");
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.NOBLE);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // When Cursed Courtier enters the battlefield, create a Cursed Role token attached to it.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new CreateRoleAttachedSourceEffect(RoleType.CURSED)));
```

Comment:
Just tried it, does not work as expected, you can select a different creature to attach the Role to it. Will try to reread your code and find the problem.

Reply:
this seems to be a preexisting bug, see #10035

Reply:
When I tried it earlier, it ended up with the aura attached to something it should not have (e.g. another creature with Cursed Courtier)
Does the change source.getFirstTarget() -> permanent.getId() fixed that to be the expected attached permanent?

Reply:
I believe so, yes